swap_meet/vendor.py

```diff
@@ -1,2 +1,99 @@
+from re import A
+from swap_meet.item import Item
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if inventory == None:
+            self.inventory =[]
+        else:
+            self.inventory = inventory
+        
+    def add(self, new_item):
+        self.inventory.append(new_item)
+        return new_item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        items=[]
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items
+
+    def swap_items(self, another_vendor, my_item, their_item):
+
+        if my_item not in self.inventory or their_item not in another_vendor.inventory:    
+            return False
+        if my_item == their_item:
+            return True
+        another_vendor.add(my_item)
+        self.add(their_item)
+        self.remove(my_item)
+        another_vendor.remove(their_item)
+        return True
+            
+    def swap_first_item(self, friend):
+        if self.inventory == [] or friend.inventory == []:
+            return False
+        self.add(friend.inventory[0])
+        friend.add(self.inventory[0])
+        self.remove(self.inventory[0])
+        friend.remove(friend.inventory[0])
+        return True
+
+    def get_best_by_category(self, category):
+        matched_category=[]
+        for item in self.inventory:
+            if item.category == category:
+                matched_category.append(item)
+
+        if len(matched_category) == 0:
+            return None
+        highest_condition = matched_category[0]
+        for item in matched_category:
+            if item.condition > highest_condition.condition:
+                highest_condition = item
+                return item
```

Comment:
Could you solve this without building the `matched_category` list (hint, combine the loops)?